synapse/handlers/room_member.py

```diff
@@ -274,6 +244,55 @@ def _local_membership_update(
 
         defer.returnValue(event)
 
+    @defer.inlineCallbacks
+    def copy_room_tags_and_direct_to_room(
+        self,
+        old_room_id,
+        new_room_id,
+        user_id,
+    ):
+        """Copies the tags and direct room state from one room to another.
+
+        Args:
+            old_room_id (str)
+            new_room_id (str)
+            user_id (str)
+
+        Returns:
+            Deferred|None
```

Comment:
ITYM `Deferred[None]` ?

Reply:
Sure, it was just copied from another method. We seem to have several different ways of writing the same thing spread across different methods?

Reply:
`Deferred|None` means: "either a Deferred (of unspecified value type), or None", which is different to what your method does

Reply:
Ahh, I see, thank you.